impl/graphsync.go

```diff
@@ -334,6 +336,14 @@ func (gsr *graphSyncReceiver) ReceiveMessage(
 	sender peer.ID,
 	incoming gsmsg.GraphSyncMessage) {
 	gsr.graphSync().responseManager.ProcessRequests(ctx, sender, incoming.Requests())
+	totalMemoryAllocated := uint64(0)
+	for _, blk := range incoming.Blocks() {
+		totalMemoryAllocated += uint64(len(blk.RawData()))
+	}
+	select {
+	case <-gsr.graphSync().responseAllocator.AllocateBlockMemory(sender, totalMemoryAllocated):
```

Comment:
If all the message receivers use the same allocator, would this still cause all other receivers to be blocked on memory allocation?

Reply:
The allocator takes a peer as a parameter cause it tracks memory allocations per peer as well as total. There's a total limit and a per peer limit. So if peer X tries to allocate above a certain amount, it will get held up seperately long before the total limit across peers gets hit.